### PR TITLE
Initial commit to add terraform modules

### PR DESCRIPTION
**Pull request by SazniMohamed**
## Overview
This pull request introduces a set of Terraform modules designed to simplify the provisioning and management of Azure resources, Azure DevOps, and Azure Active Directory within our project.

### Azure Resource Management

- Terraform modules for creating and managing Azure resources, such as virtual networks, virtual machines, storage accounts, and more.
- Infrastructure as code (IaC) solutions for improved consistency and reproducibility.
- Scalable and customizable configurations to fit various use cases.

### Azure DevOps Integration

- Modules for configuring Azure DevOps pipelines, services, and agent pools.
- Streamlined CI/CD setup for automated deployment and testing.
- Improved project efficiency and collaboration.

### Azure Active Directory (Azure AD)

- Terraform modules for setting up Azure AD tenants, users, and applications.
- Enhanced security and access control for Azure resources.
- Integration with existing authentication and authorization mechanisms.

## Why These Changes Matter
These modules simplify the process of deploying Azure resources and services, enhancing the overall project management and infrastructure provisioning experience. Here are some key benefits:

- Consistency: The IaC approach ensures consistent resource provisioning across environments.
- Productivity: Streamlined Azure DevOps integration simplifies continuous delivery and testing.
- Security: Properly configured Azure AD enhances security and access control.
- Modularity: The modular approach allows for easy customization and scalability.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,3 @@
 # azure-terraform-modules
 Terraform modules for Azure Infrastructure Resources
+Sourced from https://github.com/wso2-enterprise/azure-terraform-modules/tree/v2.88.2
```
Comment (SazniMohamed) [inline README.md:3]: remove

File: modules/azuread/AD-Group-Member/variables.tf [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+# Dissemination of any information or reproduction of any material contained
+# herein in any form is strictly forbidden, unless permitted by WSO2 expressly.
+# You may not alter or remove any copyright or other notice from copies of this content.
+#
+# --------------------------------------------------------------------------------------
+
+variable "group_object_id" {
+  description = "The object ID of the group you want to add the member to. Changing this forces a new resource to be created."
+  type        = string
+}
```
Comment (athiththan11) [inline modules/azuread/AD-Group-Member/variables.tf:15]: Shall we leave a new line between each variable definition?

File: modules/azuredevops/Kubernetes-Service-Endpoint-Kube-Config/outputs.tf [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+# herein in any form is strictly forbidden, unless permitted by WSO2 expressly.
+# You may not alter or remove any copyright or other notice from copies of this content.
+#
+# --------------------------------------------------------------------------------------
+
+output "serviceendpoint_id" {
+  depends_on = [azuredevops_serviceendpoint_kubernetes.devops_serviceendpoint_kubernetes]
+  value      = azuredevops_serviceendpoint_kubernetes.devops_serviceendpoint_kubernetes.id
+}
+output "serviceendpoint_name" {
```
Comment (athiththan11) [inline modules/azuredevops/Kubernetes-Service-Endpoint-Kube-Config/outputs.tf:16]: Shall we have a new line in between?

File: modules/azuredevops/Project/project.tf [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+#
+# --------------------------------------------------------------------------------------
+
+resource "azuredevops_project" "devops_project" {
+  name               = join("-", [var.project, var.name])
+  description        = var.description
+  visibility         = var.visibility
+  version_control    = var.version_control
+  work_item_template = var.work_item_template
+
```
Comment (athiththan11) [inline modules/azuredevops/Project/project.tf:18]:
```suggestion

```

File: modules/azuredevops/Resource-Authorization/outputs.tf
Comment (athiththan11) [inline modules/azuredevops/Resource-Authorization/outputs.tf:1]: Let's remove this

**Review by athiththan11 — CHANGES_REQUESTED** [review]

**Review by athiththan11 — APPROVED** [review]

**Review by Uvindu96 — APPROVED** [review]

**Pull request merged**